src/core/border_router/routing_manager.cpp

```diff
@@ -683,6 +683,53 @@ bool RoutingManager::IsValidOnLinkPrefix(const Ip6::Prefix &aOnLinkPrefix)
            !aOnLinkPrefix.IsMulticast();
 }
 
+bool RoutingManager::IsExternalRoutePublished() const
+{
+    NetworkData::Iterator            iterator = NetworkData::kIteratorInit;
+    NetworkData::ExternalRouteConfig routeConfig;
+    bool                             isPublished = false;
+    uint16_t                         rloc16;
+
+    rloc16 = Get<Mle::MleRouter>().GetRloc16();
+
+    // Iterate through the external routes in the network data
+    while (Get<NetworkData::Leader>().GetNextExternalRoute(iterator, routeConfig) == kErrorNone)
+    {
+        // Check if the route is stable and matches the current RLOC16
+        if (routeConfig.mStable && (routeConfig.GetRloc16() == rloc16))
+        {
+            isPublished = true;
+            break;
+        }
+    }
+
+    return isPublished;
+}
```

Comment:
What is the use-case for the new methods? Who/what module will use them?

Reply:
The new methods are planned to be used by telemetry in ot-br-posix for multi-AIL metrics

Reply:
@zesonzhang  Note that you cannot directly call methods of `RoutingManager` in ot-br-posix. You have to make them OT APIs so they can be used.